airflow/contrib/hooks/imap_hook.py

```diff
@@ -0,0 +1,262 @@
+                                                                    check_regex,
+                                                                    latest_only=True)
+        return len(mail_attachments) > 0
+
+    def retrieve_mail_attachments(self, name, mail_folder='INBOX', check_regex=False,
+                                  latest_only=False):
+        """
+        Retrieves mail's attachments in the mail folder by its name.
+
+        :param name: The name of the attachment that will be downloaded.
+        :type name: str
+        :param mail_folder: The mail folder where to look at.
+                            The default value is 'INBOX'.
+        :type mail_folder: str
+        :param check_regex: Checks the name for a regular expression.
+                            The default value is False.
+        :type check_regex: bool
+        :param latest_only: If set to True it will only retrieve
+                            the first matched attachment.
+                            The default value is False.
+        :type latest_only: bool
+        :returns: a list of tuple each containing the attachment filename and its payload.
+        :rtype: a list of tuple
+        """
+        mail_attachments = self._retrieve_mails_attachments_by_name(name, mail_folder,
+                                                                    check_regex,
+                                                                    latest_only)
+        return mail_attachments
+
+    def download_mail_attachments(self, name, local_output_directory, mail_folder='INBOX',
+                                  check_regex=False, latest_only=False):
+        """
+        Downloads mail's attachments in the mail folder by its name
+        to the local directory.
+
+        :param name: The name of the attachment that will be downloaded.
+        :type name: str
+        :param local_output_directory: The output directory on the local machine
+                                       where the files will be downloaded to.
+        :type local_output_directory: str
+        :param mail_folder: The mail folder where to look at.
+                            The default value is 'INBOX'.
+        :type mail_folder: str
+        :param check_regex: Checks the name for a regular expression.
+                            The default value is False.
+        :type check_regex: bool
+        :param latest_only: If set to True it will only download
+                            the first matched attachment.
+                            The default value is False.
+        :type latest_only: bool
+        """
+        mail_attachments = self._retrieve_mails_attachments_by_name(name, mail_folder,
+                                                                    check_regex,
+                                                                    latest_only)
+        self._download_files(mail_attachments, local_output_directory)
```

Comment:
Maybe return True/False or number of files downloaded?

Reply:
Hm.. I dont know. I personally would not return anything because it changes the filesystem. A method should either change something or return something but not both?!

Of course it should raise an exception if there is a problem when downloading and/or writing a file.